src/apiServices/ClientService.ts

```diff
@@ -46,21 +54,33 @@ export class ClientService {
         refresh: '',
     }
 
-    constructor(
-        ethAddress?: string,
-        password?: string,
-        toolName: string = 'MythXJS',
-        environment: string = 'https://api.mythx.io/v1',
-        accessToken: string = '',
-    ) {
-        this.ethAddress = ethAddress
-        this.password = password
-        ClientService.MYTHX_API_ENVIRONMENT = environment
-        this.authService = new AuthService(ethAddress, password)
-        ;(this.toolName = toolName), (ClientService.jwtTokens.access = accessToken)
-        if (accessToken) {
-            ClientService.jwtTokens.access = accessToken
+    constructor(clientConfig: ClientConfig)
+    constructor(clientConfig: any) {
+        if (clientConfig.username && clientConfig.apiKey) {
+            this.ethAddress = clientConfig.username
+            ClientService.MYTHX_API_ENVIRONMENT = clientConfig.environment || 'https://api.mythx.io/v1'
+            this.authService = new AuthService(this.ethAddress)
+            ;(this.toolName = clientConfig.toolName || 'MythXJS'),
```

Comment:
Is this valid syntax? Can you elaborate on the semicolon and the list notation here?

Reply:
Great question! :D This was part of the existing code base that I didn't write, so I had to go look it up and ended up leaning new things as well.

### The semi-column
The `;` is there because the expression after it starts with a `(` and JS could interpret that as being the continuation of the previous line, for instance if `new AuthService(this.ethAddress)` returned a function, then the following line without `;` would be calling that function. More on this here https://stackoverflow.com/questions/3721784/when-is-it-dangerous-to-line-break-after-a-close-parenthesis-in-javascript

### The comma separated statements
In JS you can bundle statements together using commas. This is usually useful for things like for statements where you want to initialize more than 1 variable `for(a = 0, b=10; b<20; b++ ) { ... }` . More on that https://stackoverflow.com/questions/8506305/js-statement-separator-with-commas